modules/account/src/main/kotlin/edu/stanford/spezi/module/account/cred/manager/CredentialManagerAuth.kt

```diff
@@ -0,0 +1,143 @@
+package edu.stanford.spezi.module.account.cred.manager
+
+import android.content.Context
+import androidx.credentials.CreateCredentialResponse
+import androidx.credentials.CreatePasswordRequest
+import androidx.credentials.CredentialManager
+import androidx.credentials.CustomCredential
+import androidx.credentials.GetCredentialRequest
+import androidx.credentials.GetCredentialResponse
+import androidx.credentials.PasswordCredential
+import androidx.credentials.exceptions.GetCredentialException
+import com.google.android.libraries.identity.googleid.GetGoogleIdOption
+import com.google.android.libraries.identity.googleid.GoogleIdTokenCredential
+import com.google.android.libraries.identity.googleid.GoogleIdTokenParsingException
+import edu.stanford.spezi.core.logging.speziLogger
+import edu.stanford.spezi.core.navigation.DefaultNavigationEvent
+import edu.stanford.spezi.core.navigation.Navigator
+import edu.stanford.spezi.module.account.AccountNavigationEvent
+import edu.stanford.spezi.module.account.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class CredentialManagerAuth @Inject constructor(
+    private val credentialManager: CredentialManager,
+    private val firebaseAuthManager: FirebaseAuthManager,
+    private val navigator: Navigator,
+) : SignInHandler {
+    private val logger by speziLogger()
+
+    override fun handleSignIn(
```

Comment:
In addition to what Paul has proposed, some API design related comments.

I would mark the function as suspend and there is no need to pass the scope as a parameter. Also you can inject `@ApplicationContext private val context: Context` in the constructor, hence no for context parameter as well.

Do we then want to expose `GoogleIdTokenCredential` in the public api? Can't we just return a `Result<Type>` from this function instead? Also I would decouple the navigation actions from this component and solely return an appropriate result type (You can introduce a sealed class with possible results) and handle navigation in view model instead